android-beacon/src/main/java/com/novoda/beacon/MessageService.java

```diff
@@ -0,0 +1,90 @@
+package com.novoda.beacon;
+
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.IBinder;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.nearby.Nearby;
+import com.google.android.gms.nearby.messages.Strategy;
+import com.google.android.gms.nearby.messages.SubscribeOptions;
+
+public class MessageService extends Service {
+
+    private static final String TAG = "!!!";
+
+    private GoogleApiClient googleApiClient;
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        googleApiClient = new GoogleApiClient.Builder(this)
+                .addApi(Nearby.MESSAGES_API)
+                .addConnectionCallbacks(connectionCallbacks)
+                .addOnConnectionFailedListener(onConnectionFailedListener)
+                .build();
+
+        googleApiClient.connect();
+        return Service.START_NOT_STICKY;
+    }
+
+    private final GoogleApiClient.ConnectionCallbacks connectionCallbacks = new GoogleApiClient.ConnectionCallbacks() {
+        @Override
+        public void onConnected(@Nullable Bundle bundle) {
+            Log.d(TAG, "onConnectionConnected");
+            subscribe(googleApiClient);
+        }
+
+        private void subscribe(GoogleApiClient googleApiClient) {
```

Comment:
I'm easy, when the methods are inside the anon impls they're easier to extract into a class and it shows they're not _realllyyy_ part of the outer